netCDF4/_netCDF4.pyx

```diff
@@ -1186,8 +1257,17 @@ cdef _get_types(group):
                     #print 'WARNING: unsupported VLEN type, skipping...'
                     continue
                 vltypes[name] = vltype
-                pass
-    return cmptypes, vltypes
+            elif classp == NC_ENUM: # an enum type
+                name = namstring.decode(default_encoding,unicode_error)
+                # read the Enum type info from the file,
+                # create a EnumType instance from it.
+                try:
+                    enumtype = _read_enum(group, xtype)
+                except KeyError:
+                    #print 'WARNING: unsupported Enum type, skipping...'
```

Comment:
I used to raise an error for Compound and VLEN types when the read failed, but some users suggested that it would be better to just proceed and read the rest of the file silently ignoring the variables and data types that could not be handled.  Since the potential complexity of HDF5 data types is almost limitless (compound types including vlens of enums..) it is almost certain that there will be files we can't read.